File: Contains Duplicate/containsDuplicate.java
/**
 * https://leetcode.com/problems/contains-duplicate/description/
time complexity: O(l) where l is the length of the nums array
space complexity: O(l) where l is the length of the nums array
approach: iterate over the nums array and add number to the hashset if it is not already present in it. if the number is already present in the hashset, return true
*/
import java.util.HashSet;
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for(int i : nums)
        {
            if(set.contains(i)) return true;
            set.add(i);
        }
        return false;
    }
}

File: Contains Duplicate/contains_duplicate.py
'''
https://leetcode.com/problems/contains-duplicate/description/
time complexity: O(n) where n is the length of the nums list
space complexity: O(n) where n is the length of the nums list
approach: iterate over the numbers of the nums list and add them to a numSet list if the current number is not present in numSet, else return True
'''
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        numSet = set()
        for currNum in nums:
            if currNum in numSet:
                return True
            numSet.add(currNum)
        return False